### ISSUE TITLE
Error: Internal assertion failed [] when I try to add new Control Layer.

### ISSUE THREAD

**Issue by pendave** [Acly/krita-ai-diffusion#846]
I met this error when I add new Control Layer as the Filter Mask is selected and highlighted:
![image](https://github.com/Acly/krita-ai-diffusion/assets/1331881/049a4f13-194a-448d-9a8c-b84c91a2c6c1)

**Comment by Acly** [comment]
Can you check client.log for the full error?

**Comment by pendave** [comment]
It happend when the Filter Mask is selected and highlighted:
![image](https://github.com/Acly/krita-ai-diffusion/assets/1331881/b8236196-0d76-4795-a7d1-6b742a6e6fa7)

Then the new added Control Layer is
![image](https://github.com/Acly/krita-ai-diffusion/assets/1331881/e3251a57-a146-4992-88f7-d95e61717b9e)

**Comment by Acly** [comment]
Hm it works fine for me. Are you using version 1.18.1?
Also if you can find the error somewhere at bottom of client.log file this will help [[Log files]](https://github.com/Acly/krita-ai-diffusion/wiki/Common-Issues#log-files)

**Comment by pendave** [comment]
> Hm it works fine for me. Are you using version 1.18.1? Also if you can find the error somewhere at bottom of client.log file this will help [[Log files]](https://github.com/Acly/krita-ai-diffusion/wiki/Common-Issues#log-files)

```Python
Traceback (most recent call last):
  File "C:\Users\pen_d\AppData\Roaming\krita\pykrita\ai_diffusion\model.py"
    return await coro
  File "C:\Users\pen_d\AppData\Roaming\krita\pykrita\ai_diffusion\model.py"
    new_input = await self._model._generate_live(self._last_input)
  File "C:\Users\pen_d\AppData\Roaming\krita\pykrita\ai_diffusion\model.py"
    cond
  File "C:\Users\pen_d\AppData\Roaming\krita\pykrita\ai_diffusion\region.py"
    control=[c.to_api(bounds) for c in list(root.control) + parent_control]
  File "C:\Users\pen_d\AppData\Roaming\krita\pykrita\ai_diffusion\region.py"
    control=[c.to_api(bounds) for c in list(root.control) + parent_control]
  File "C:\Users\pen_d\AppData\Roaming\krita\pykrita\ai_diffusion\control.py"
    image = layer.get_pixels(bounds)
  File "C:\Users\pen_d\AppData\Roaming\krita\pykrita\ai_diffusion\layer.py"
    assert data is not None and data.size() >= bounds.extent.pixel_count * 4
AssertionError
2024-06-20 16:32:08
Traceback (most recent call last):
  File "C:\Users\pen_d\AppData\Roaming\krita\pykrita\ai_diffusion\model.py"
    return await coro
  File "C:\Users\pen_d\AppData\Roaming\krita\pykrita\ai_diffusion\model.py"
    new_input = await self._model._generate_live(self._last_input)
  File "C:\Users\pen_d\AppData\Roaming\krita\pykrita\ai_diffusion\model.py"
    cond
  File "C:\Users\pen_d\AppData\Roaming\krita\pykrita\ai_diffusion\region.py"
    control=[c.to_api(bounds) for c in list(root.control) + parent_control]
  File "C:\Users\pen_d\AppData\Roaming\krita\pykrita\ai_diffusion\region.py"
    control=[c.to_api(bounds) for c in list(root.control) + parent_control]
  File "C:\Users\pen_d\AppData\Roaming\krita\pykrita\ai_diffusion\control.py"
    image = layer.get_pixels(bounds)
  File "C:\Users\pen_d\AppData\Roaming\krita\pykrita\ai_diffusion\layer.py"
    assert data is not None and data.size() >= bounds.extent.pixel_count * 4
AssertionError
```

**Comment by Acly** [comment]
Thanks. I only tried adding the layer, not generating. Since you were in active Live mode it tried to use the filter mask as control layer to generate an image. Looks like that doesn't work, masks can't be used as image source.

The plugin should probably select the parent and hide masks from the dropdown to avoid the error.

**Comment by Acly** [comment]
Fixed in [v1.19.0](https://github.com/Acly/krita-ai-diffusion/releases/tag/v1.19.0)

**Comment by orkutmuratyilmaz** [comment]
I'm having the same issue with,
Krita 5.2.9
Generative AI for Krita 1.31.2

Here is my error log:

```
Traceback (most recent call last):

  File "/home/user/.local/share/krita/pykrita/ai_diffusion/model.py", line 530, in update_preview
    self.show_preview(selection.job, selection.image)

  File "/home/user/.local/share/krita/pykrita/ai_diffusion/model.py", line 549, in show_preview
    self._layer.write_pixels(image, bounds)

  File "/home/user/.local/share/krita/pykrita/ai_diffusion/layer.py", line 177, in write_pixels
    self._node.setPixelData(blank.data, *layer_bounds)

  File "/home/user/.local/share/krita/pykrita/ai_diffusion/image.py", line 486, in data
    ptr = ensure(self._qimage.constBits(), "Accessing data of invalid image")

  File "/home/user/.local/share/krita/pykrita/ai_diffusion/util.py", line 99, in ensure
    assert value is not None, msg or "a value is required"

AssertionError: Accessing data of invalid image
```
